component/main.jsonnet

```diff
@@ -1,10 +1,23 @@
 // main template for nfs-client-provisioner
 local kap = import 'lib/kapitan.libjsonnet';
 local kube = import 'lib/kube.libjsonnet';
+local sc = import 'lib/storageclass.libsonnet';
 local inv = kap.inventory();
 // The hiera parameters for the component
 local params = inv.parameters.nfs_client_provisioner;
 
+local storageclass = sc.storageClass(params.storageClass.name) {
```

Comment:
In future, we might have the need to configure more than one storage class (different behavior on delete and or mount options etc.). I would model this as a dictionary containing one or several storage classes. The name would become the a key in the dictionary.

In addition, make this a transparent pass through Instead of hard wiring those fields. Layer different levels of precedence on top of each other (result of the StorageClass function, component wide defaults, values configured per StorageClass). Comment assumes the ability to configure multiple StorageClasses.

Reply:
This is now implemented, and the component documentation has been updated.